core/src/main/scala/org/apache/spark/internal/config/Python.scala

```diff
@@ -50,4 +50,9 @@ private[spark] object Python {
     .version("2.4.0")
     .bytesConf(ByteUnit.MiB)
     .createOptional
+
+  val PYTHON_GATEWAY_CONNECT_TIMEOUT = ConfigBuilder("spark.python.gateway.connectTimeout")
+    .version("3.1.0")
```

Comment:
I think I would mark this as `.internal()`.

Reply:
Added.